### PR TITLE
Harrison: add visualization utils

### PR DESCRIPTION
**Pull request by hwchase17**
This adds functionality for taking raw YOLO model outputs and plotting them on top of an image

### CODE REVIEW

File: keras_yolo3/vizualization.py [hunk 1]
```diff
@@ -0,0 +1,130 @@
+import colorsys
+import pkg_resources
+
+from PIL import ImageFont, ImageDraw
+import numpy as np
+import tensorflow as tf
+
+from keras_yolo3.model import yolo_eval
+
+filename = pkg_resources.resource_filename('keras_yolo3',
```
Comment (kojino) [inline keras_yolo3/vizualization.py]: something like `font_file_name` or `font_path`?

File: keras_yolo3/utils.py [hunk 2]
```diff
@@ -119,3 +119,13 @@ def get_random_data(annotation_line, input_shape, random=True, max_boxes=20, jit
         box_data[:len(box)] = box
 
     return image_data, box_data
+
+
+def load_image(image_path):
+    """Helper to load image and return both PIL object and array."""
```
Comment (ebalkans) [inline keras_yolo3/utils.py]: what's in the image_data array? maybe add that in the method description?

File: keras_yolo3/vizualization.py [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+    colors = list(
+        map(lambda x: (int(x[0] * 255), int(x[1] * 255), int(x[2] * 255)),
+            colors))
+    np.random.seed(10101)  # Fixed seed for consistent colors across runs.
+    np.random.shuffle(colors)
+    return colors
+
+
+def _get_default_anchors():
+    """Get default anchors."""
```
Comment (retagaine) [inline keras_yolo3/vizualization.py]: what are the anchors for?

**Comment by retagaine** [comment]
also, how come there are 8 commits which just say `--no-edit`?

File: keras_yolo3/utils.py [hunk 4]
```diff
@@ -119,3 +119,13 @@ def get_random_data(annotation_line, input_shape, random=True, max_boxes=20, jit
...
+    original_image = Image.open(image_path)
+    image_data = letterbox_image(original_image, (416, 416))
+    image_data = np.array(image_data, dtype='float32')
+    image_data /= 255.
+    image_data = np.expand_dims(image_data, 0)  # Add batch dimension.
+    return original_image, image_data
```
[the remark this answers is not in the document]
Comment (hwchase17) [reply keras_yolo3/utils.py]: original image is just a JPEG image so im sure what the benefits of doing the work to divide by 255 are...

[the remark this answers is not in the document]
Comment (kojino) [reply keras_yolo3/utils.py]: seems inconsistent that two images are returned with different data types

File: keras_yolo3/vizualization.py [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+        label = '{} {:.2f}'.format(predicted_class, score)
+        draw = ImageDraw.Draw(image)
+        label_size = draw.textsize(label, font)
+
+        top, left, bottom, right = box
+        top = max(0, np.floor(top + 0.5).astype('int32'))
+        left = max(0, np.floor(left + 0.5).astype('int32'))
+        bottom = min(image.size[1], np.floor(bottom + 0.5).astype('int32'))
+        right = min(image.size[0], np.floor(right + 0.5).astype('int32'))
+        print(label, (left, top), (right, bottom))
```
[the remark this answers is not in the document]
Comment (hwchase17) [reply keras_yolo3/vizualization.py]: label has score and class!

File: keras_yolo3/utils.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (hwchase17) [reply keras_yolo3/utils.py]: but they are already two very different types... one is the original image (not altered), the other is the image data as ready to be fed into the model...

**Pull request merged**